src/main/java/com/liferay/ide/idea/ui/modules/LiferayModuleNameLocationComponent.java

```diff
@@ -104,7 +108,14 @@ protected void textChanged(DocumentEvent event) {
 					String moduleName = _getModuleName();
 
 					if ((path.length() > 0) && !Comparing.strEqual(moduleName, namePathComponent.getNameValue())) {
-						path += "/" + _getTargetFolderName() + "/" + moduleName;
+						String buildType = _getProjectBuildType();
+
+						if (buildType.equals(LiferayProjectType.LIFERAY_MAVEN_WORKSPACE)) {
+							path += "/" + moduleName;
+						}
+						else {
+							path += "/" + _getTargetFolderName() + "/" + moduleName;
```

Comment:
this need to check gradle worksapce plugin version. If its version is greater than 2.5.0 then we need to support user can customized all possible values for module directory and all wars need to put into module folder defined by liferay.workspace.modules.dir. At the same time, this property user can putl *. Most detail logic you need to reference IDE-4823 implementation